PrebidMobile/PrebidMobile-core/src/test/java/org/prebid/mobile/OpenRtbMergerTest.java

```diff
@@ -0,0 +1,134 @@
+package org.prebid.mobile;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.jsonpatch.diff.JsonDiff;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.prebid.mobile.test.utils.ResourceUtils;
+
+public class OpenRtbMergerTest {
+
+    @Test
+    public void merge_emptyOpenRtb() throws JSONException {
+        String request = "{}";
+        String openRtb = "";
+
+        JSONObject mergedJson = merge(request, openRtb);
+
+        assertJsonEquals("{}", mergedJson.toString());
+    }
+
+    @Test
+    public void merge_wrongJson() throws JSONException {
+        String request = "{}";
+        String openRtb = "not a json string";
+
+        JSONObject mergedJson = merge(request, openRtb);
+
+        assertJsonEquals("{}", mergedJson.toString());
+    }
+
+    @Test
+    public void mergeSensitiveData_emptyRequest_emptyResult() throws JSONException {
+        String request = "{}";
+        String openRtb = fromResources("sensitive_data_fake.json");
+
+        JSONObject mergedJson = merge(request, openRtb);
+
+        assertJsonEquals(fromResources("sensitive_data_empty.json"), mergedJson.toString());
+    }
+
+    @Test
+    public void mergeSensitiveData_fullRequest_requestFieldsAreNotChanged() throws JSONException {
+        String request = fromResources("sensitive_data_real.json");
+        String openRtb = fromResources("sensitive_data_fake.json");
+
+        JSONObject mergedJson = merge(request, openRtb);
+
+        assertJsonEquals(request, mergedJson.toString());
+    }
+
+    @Test
+    public void merge_differentTypes() throws JSONException {
+        String request = "{}";
+        String openRtb = fromResources("merge_all_types.json");
```

Comment:
Please don't use files. It's much more convenient to see the JSONs right in the tests.